### DISCUSSION TITLE
Does MPD support HTTP Live Streaming (HLS) URIs?

### DISCUSSION THREAD

**Discussion by jm718** [MusicPlayerDaemon/MPD#2083] [Q&A] [answered]
### Edit: Well this is embarrassing...
I forgot to do: 'mpc add url', before 'mpc play'.

I still get no sound out of the system, but the verbose mpd log makes it clear ffmeg is called and treating the stream as hls.  Will close this question and investigate further.  Sorry to waste peoples time.

### Original post:

When I enter "mpc play https://mediaserviceslive.akamaized.net/hls/live/2038311/newsradio/index.m3u8" I get "error parsing song numbers from: https://mediaserviceslive.akamaized.net/hls/live/2038311/newsradio/index.m3u8".  
I have MPD loaded on two machines, a headless Raspberry pi and Ubuntu, both mpd instances fail to play the url, on both machines ffplay is successful.

Neither machine repositories have the latest releases of mpd (0.23.14 and 0.23.12).  I have failed to compile the lastest mpd on the Raspberry pi (rp locks up), so cannot test the latest release, however the mpd --version output lists "hls+https:" under "protocols" for both versions I have, which I presume means HLS over HTTPS.

So before I spend more time troubleshooting the compile errors: should mpd be able to play a hls stream?  Any thoughts on what configuration may need to be changed?
Output of mpd --version attached
Cheers
(PS: first post, be gentle)
[mpdVersion.txt](https://github.com/user-attachments/files/16278908/mpdVersion.txt)

**Comment by skidoo23** [comment]
Maybe some issue with nested .m3u8? At least https://mediaserviceslive.akamaized.net/hls/live/2038311-b/newsradio/masterhq.m3u8 (URL found inside curl'ed .m3u8 file. Note '-b') works with recent MPD master from git.

**Reply by jm718** [reply]
Thanks for your response.  Sounds like it is worth persevering with compiling the up to date version of mpd.

**Reply by jm718** [reply, marked as answer]
I forgot to do: 'mpc add url', before 'mpc play'.  I still get no sound out of the system, but the verbose mpd log makes it clear ffmeg is called and treating the stream as hls. Will close this question and investigate further. Sorry to waste peoples time.

**Reply by MaxKellermann** [reply]
Next time, read https://www.musicpd.org/help/ first
